From 9b1256cdef301abda6d77fd82164f3eddc1f54a1 Mon Sep 17 00:00:00 2001
From: Lin Yujun <linyujun809@huawei.com>
Date: Sun, 23 Apr 2023 03:34:04 +0000
Subject: [PATCH 0387/1646] [Huawei] Fix EXPORT symbol __SCT__perf_lopwr_cb
 build warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188709

--------------------------------

a9cbff64a895 ("kabi: Fix kabi breakage caused by commit d5616bac7ada.")
use __GENKSYMS__ to fix the kabi change introduced by
fb5f22f8054d ("perf/x86/amd: Add idle hooks for branch sampling")

However, use __GENKSYMS__ control linux/static_call.h makes
the symbol "__SCT__perf_lopwr_cb" invisible when checking the CRC
value, which introduces the following compilation warning：
WARNING: modpost: EXPORT symbol "__SCT__perf_lopwr_cb" [vmlinux] version generation failed, symbol will not be versioned.

use __GENKSYMS__ to control perf_lopwr_cb() to fix this warning.
Fixes: a9cbff64a895 ("kabi: Fix kabi breakage caused by commit d5616bac7ada.")

Signed-off-by: Lin Yujun <linyujun809@huawei.com>
---
 arch/x86/include/asm/perf_event.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/include/asm/perf_event.h b/arch/x86/include/asm/perf_event.h
index 77b92b234f83..6cb26772f74b 100644
--- a/arch/x86/include/asm/perf_event.h
+++ b/arch/x86/include/asm/perf_event.h
@@ -552,12 +552,14 @@ static inline void intel_pt_handle_vmx(int on)
  */
 extern void perf_amd_brs_lopwr_cb(bool lopwr_in);
 
+#ifndef __GENKSYMS__
 DECLARE_STATIC_CALL(perf_lopwr_cb, perf_amd_brs_lopwr_cb);
 
 static inline void perf_lopwr_cb(bool lopwr_in)
 {
 	static_call_mod(perf_lopwr_cb)(lopwr_in);
 }
+#endif
 
 #endif /* PERF_NEEDS_LOPWR_CB */
 
-- 
2.17.1

